go/logging.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+)
+
+func loggingMiddleware(logger log.Logger) ServiceMiddleware {
+	return func(next StringService) StringService {
+		return logmw{logger, next}
+	}
+}
+
+type logmw struct {
+	logger log.Logger
+	StringService
+}
+
+func (mw logmw) InitialUppercase(ctx context.Context, s string) (output string, err error) {
+	defer func(begin time.Time) {
+		_ = mw.logger.Log(
+			"method", "uppercase",
+			"telepresenceHeader", ctx.Value("x-telepresence-id").(string),
```

Comment:
This may just be my Java roots talking here, but could this header string value be made into something like a constant or enum? (I always get nervous when I see a header name written out in full in multiple places :) )

Reply:
Yeah, it's been awhile since I've worked in Go so let me double check how global consts work.